accepted/2020/platform-checks/platform-checks.md

```diff
@@ -0,0 +1,450 @@
+  resulted in Windows-specific APIs being added to types that are otherwise
+  cross-platform. A good example are file system and threading types that have
+  methods for setting the Windows access control lists (ACL). While we could
+  provide them elsewhere, this would unnecessarily break a lot of code that
+  builds Windows-only applications, e.g. WinForms and WPF apps.
+
+* **OS platforms are typically targeted using the latest SDK**. Even when the
+  project is platform specific, such as `net5.0-ios` or `net5.0-android`, it's
+  not necessarily clear that the called API is actually available. For .NET
+  Framework, we used *reference assemblies* which provides a different set of
+  assemblies for every version. However, for OS bindings this approach is not
+  feasible. Instead, applications are generally compiled against the latest
+  version of the SDK with the expectations that callers have to check the OS
+  version if they want to run on older versions. See [Minimum OS
+  Version][os-minimum-version] for more details.
+
+.NET has always taken the position that providing platform-specific
+functionality isn't a bug but a feature because it empowers developers to build
+applications that feel native for the user. That's why .NET has rich interop
+facilities such as P/Invoke.
+
+At the same time, developers really dislike the notion of "write once and test
+everywhere" because it results in a delayed feedback loop.
+
+Fortunately .NET has also taken the position that developer productivity is key.
+This document proposes:
+
+1. A mechanism to annotate APIs as being platform-specific, including the
+   version they got introduced in, the version they got removed in, and the
+   version they got obsoleted in.
+
+2. A Roslyn analyzer that informs developers when they use platform-specific
+   APIs from call sites where the API might not be available.
+
+This work draws heavily form the experience of the [API Analyzer] and improves
+on it by making it a first-class platform feature.
+
+## Scenarios and User Experience
+
+### Lighting up on later versions of the OS
+
+Miguel is building Baby Shark, a popular iOS application. He started with .NET
+that supported iOS 13 but Apple just released iOS 14, which adds the great
+`NSFizBuzz` API that gives his app the extra pop.
+
+After upgrading the app and writing the code, Miguel decides that while the
+feature is cool, it doesn't warrant cutting of all his customers who are
+currently still on iOS 13. So he edits the project file and sets
+`TargetPlatformMinVersion` to 13.
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0-ios14.0</TargetFramework>
+    <TargetPlatformMinVersion>13.0</TargetPlatformMinVersion>
+  </PropertyGroup>
+
+  ...
+
+</Project>
+```
+
+After doing this, he gets a diagnostic in this code:
+
+> 'NSFizzBuff' requires iOS 14 or later.
+
+```C#
+static void ProvideExtraPop()
```

Comment:
Nit: Current .NET coding standards says you should be explicit with the access modifier. We should be good examples in our docs.